backend/api/routes/ai_game.py

```diff
@@ -42,21 +45,33 @@ def get_agent(self, white_model_name: str = "PPO", black_model_name: str = "PPO"
                 black_model = self.BLACK_PPO_PATH
             case "DQN":
                 black_model = self.BLACK_DQN_PATH
+            case "A2C":
+                black_model = self.BLACK_A2C_PATH
             case _:
                 black_model = self.BLACK_PPO_PATH
 
         return PPOChess(self.env, self.get_ppo(), 1, 32, "", white_model, black_model)
 
     def play_game(self, agent, episode) -> AIGameResponse:
-        response = AIGameResponse(game=[], statistics=[], possibles=[], source_pos=[], action_mask=[])
+        response = AIGameResponse(game=[], statistics=[], possibles=[], source_pos=[], action_mask=[], winner="")
 
         response.game.append(agent.env.aow_board.get_numeric_board().tolist())
         response.statistics.append({"rewards": [0, 0], "infos": [[], []], "end": False})
 
         done = False
+        info = None
         while not done:
-            done, _ = agent.take_action(agent.env.aow_logic.turn, episode)
-            response.statistics.append({"rewards": _[1], "infos": _[7], "end": done})
+            done, info = agent.take_action(agent.env.aow_logic.turn, episode)
+            response.statistics.append({"rewards": info[1], "infos": info[7], "end": done})
             response.game.append(agent.env.aow_board.get_numeric_board().tolist())
+
+        if CHECK_MATE_WIN in info[7][0]:

```

Comment:
is het een idee dat je hier een methode uit haalt waar je infos aan mee geeft. en uiteindelijk krijg je of een lege string terug of, blackwin, whitewin, draw. dan blijft de play_game methode wat schoner